src/raftstore/store/fsm/peer.rs

```diff
@@ -794,6 +797,28 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
             self.fsm.has_ready = true;
         }
 
+        // If lease expired, we will send a noop read index to renew lease.
+        if self.fsm.peer.is_leader() && LeaseState::Expired == self.fsm.peer.inspect_lease() {
```

Comment:
`inspect_lease` makes a syscall to get the current time, we should avoid it as much as possible.

Also, could you add a test to ensure it works as expected?

Reply:
So is it expensive when there are many regions? which is more recommended with hibernate regions?